lib/fencing/st_rhcs.c

```diff
@@ -30,37 +32,55 @@ stonith__list_rhcs_agents(stonith_key_value_t **devices)
 {
     // Essentially: ls -1 @sbin_dir@/fence_*
 
-    int count = 0;
+    int count = 0, i;
     struct dirent **namelist;
-    int file_num = scandir(RH_STONITH_DIR, &namelist, 0, alphasort);
-
-    if (file_num > 0) {
+    const int file_num = scandir(RH_STONITH_DIR, &namelist, 0, alphasort);
+
+#if (!defined(_POSIX_C_SOURCE) || _POSIX_C_SOURCE < 200809L) \
+        && (!defined(O_SEARCH) || !defined(O_PATH))
```

Comment:
I think you mean && for the last, otherwise both O_SEARCH and O_PATH must be defined to get past here